hack/generator/pkg/astmodel/oneof_type.go

```diff
@@ -24,13 +24,15 @@ type OneOfType struct {
 	types TypeSet
 }
 
+var _ Type = &OneOfType{}
+
 // MakeOneOfType is a smart constructor for a  OneOfType,
 // maintaining the invariants. If only one unique type
 // is passed, the result will be that type, not a OneOf.
 func MakeOneOfType(types ...Type) Type {
```

Comment:
Same comment as the `AllOf` type - unsure if changing this name makes sense or not - although I do note that in other instances such as the IdFactory when we return a ptr even if the actual specified type is an interface we still say `New`